config/routes.rb

```diff
@@ -1,8 +1,11 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
-  # Define your application routes per the DSL in https://guides.rubyonrails.org/routing.html
-  # resources :users, only: [:new, :create]
-  # Defines the root path route ("/")
-  # root "articles#index"
+  root 'users#index'
+
+  get '/users', to: 'users#index', as: 'users' # users_path
+  get '/users/:id', to: 'users#show', as: 'user' # user_path
+  get '/users/:id/posts', to: 'users#posts', as: 'user_posts' # user_posts_path
+  get '/users/:id/posts/:post_id', to: 'posts#show', as: 'user_post' # user_post_path
+
```

Comment:
- [ ] [Optional] I would suggest using the below shorthand code for the above routes. It will make the code more clean and professional.

```ruby
resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end